zomes/chat/src/pagination_helper.rs

```diff
@@ -0,0 +1,119 @@
+//! TODO: Document how to use this crate in general
+//!
+//!
+//!
+use crate::error::ChatResult;
+use chrono::{DateTime, Datelike, Duration, NaiveDateTime, Timelike, Utc};
+use hdk::prelude::*;
+use std::ops::Sub;
+
+///
+pub fn get_batch_links(
+    base: Path,
+    earlier_than: Timestamp,
+    target_count: usize,
+) -> ChatResult<Vec<Link>> {
+    let mut links: Vec<Link> = Vec::new();
+    let mut next_path = timestamp_into_path(base.clone(), earlier_than, None)?;
+
+    loop {
+        debug!("Next path: {:?}", next_path.clone());
+        // Confirm the path exists
+        if next_path.exists()? {
+            debug!("next path exists");
+            // Get the actual hash we are going to pull the messages from
+            let channel_entry_hash = next_path.hash()?;
+            // Get the message links on this channel
+            let nm = &mut get_links(channel_entry_hash.clone(), None)?.into_inner();
+            let nm_check = nm.is_empty();
```

Comment:
If this is true, it indicates that this chunk has messages in it, but the links haven't been gossiped to us yet, right? Why should we break out of the loop in that case? It seems like we should keep looking until we reach our target count.